Assets/Scripts/Item/ItemInventory.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using UnityEngine;
+
+[Serializable]
+[CreateAssetMenu(fileName = "ItemInventory", menuName = "ScriptableObject/Item/ItemInventory")]
+public class ItemInventory : ScriptableObject
+{
+    [SerializeField] private List<Item> itemList;
+    private HashSet<Item> itemSet;
+    private Dictionary<string, Item> itemDict;

```

Comment:
```suggestion
    private Dictionary<string, int> itemDict;
```
せっかくアイテムリストがあるならindexでいいんじゃない？